cpp/src/communicator/ucxx.cpp

```diff
@@ -1007,8 +1014,9 @@ constexpr ::ucxx::TagMask UserTagMask{std::numeric_limits<uint32_t>::max()};
 std::shared_ptr<::ucxx::Endpoint> UCXX::get_endpoint(Rank rank) {
     Logger& log = logger();
     try {
+        auto ep = shared_resources_->get_endpoint(rank);
         log.trace("Endpoint for rank ", rank, " already available, returning to caller");
-        return shared_resources_->get_endpoint(rank);
+        return ep;
```

Comment:
Nit: revert this change. The reasoning is the logging will be omitted if `shared_resources_->get_endpoint(rank)` raises an exception, plus I think the original was more straightforward.

Reply:
Ah yeah, you're right, my bad.